### PR TITLE
Add a rustls-webpki harness

### LINKED ISSUE #88: Rustls webpki test harness?
Hi folks, great project :lock: :hammer: 

I saw that https://github.com/trailofbits/x509-limbo/pull/80 added a test harness for `briansmith/webpki`. Would you be willing to do similar for [rustls/webpki](https://github.com/rustls/webpki)?

The [rustls](https://github.com/rustls/rustls) project uses this fork instead of the original repository since Rustls [v0.21.0](https://github.com/rustls/rustls/releases/tag/v%2F0.21.0). It has some capabilities not present in the original repository (like IP address subject support, and CRL based revocation checking).

We've invested in running other testing regimens like [BetterTLS](https://bettertls.com/) in-repo but are always keen for more coverage.

### PR DESCRIPTION
**Pull request by woodruffw**
I got a little bored on a flight and added this. ~~Still needs to be hooked up to the CI/CD.~~ Done.

CC @cpu -- you might/should be able to reuse bits of this for your own in-tree runner 🙂 

Closes #88.

### CODE REVIEW

File: harness/rust-rustls/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        Some(pn) => match pn.kind {
+            PeerKind::Dns => webpki::SubjectNameRef::DnsName(
+                webpki::DnsNameRef::try_from_ascii_str(&pn.value)
+                    .expect("invalid expected DNS name"),
+            ),
+            PeerKind::Ip => {
+                // Very dumb: rustls-webpki doesn't allow compressed IPv6 string representations,
+                // so we need to round-trip through `std::net::IpAddr`. This in turn requires
+                // us to round-trip through `webpki::IpAddr` and perform a leak, since
+                // we have no outliving reference to borrow against.
```
Comment (cpu) [inline harness/rust-rustls/src/main.rs:124]: FWIW, as of rustls/webpki 0.102.0-alpha.7 the compressed repr should work. We'll probably have a final release with that code in the next couple of weeks.

**Comment by cpu** [comment]
Thank you, this is great!

File: harness/rust-rustls/src/main.rs (hunk 1 above)
Comment (woodruffw) [reply harness/rust-rustls/src/main.rs:124]: Awesome, thank you! We'll keep this harness up-to-date with each version as it goes out of alpha 🙂 

(Also for the record, the "very dumb" is referring to me leaking to hack around the lifetimes here, not your APIs.)

Comment (cpu) [reply harness/rust-rustls/src/main.rs:124]: > Also for the record, the "very dumb" is referring to me leaking to hack around the lifetimes here, not your APIs

Haha yeah, that's the context I read it in too but I appreciate the clarification :-D

**Pull request merged**